doc/configuration.rst

```diff
@@ -1327,6 +1331,56 @@ For more information on CircleCI integration, peruse the related
 `CircleCI doc <https://circleci.com/docs/2.0/collect-test-data/#metadata-collection-in-custom-test-steps>`__
 and `blog post <https://circleci.com/blog/how-to-output-junit-tests-through-circleci-2-0-for-expanded-insights/>`__.
 
+Parametrized ``pytest`` cases
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The JUnit XML file can also be parsed manually and used to generate parametrized
+test cases with ``pytest``. For example, to test computation times for your
+project's documentation, define a new test module
+
+.. code-block:: shell
+
+    <your_project>/tests/test_execution_times.py
+
+and include the following code block. Be sure to configure the options
+``MAX_EXECUTION_TIME``, ``OUT_DIR``, and ``XML_FILE`` to suit your project.
+
+.. code-block:: python
+
+    from pathlib import Path
+    from xml.etree.ElementTree import parse
+
+    import pytest
+
+    MAX_EXECUTION_TIME = 5.0  # Seconds
+
+    # Full xml path relative to this test module
+    OUT_DIR = Path(__file__).parents[2] / "doc" / "_build" / "html"
+    XML_FILE =  OUT_DIR / "sphinx-gallery" / "junit-results.xml"
+    assert XML_FILE.is_file()
+
+    xml_root = parse(XML_FILE).getroot()
+    test_cases = [dict(case.attrib) for case in xml_root.iterfind("testcase")]
+    test_ids = [case["classname"] for case in test_cases]
+
+    @pytest.mark.parametrize("testcase", test_cases, ids=test_ids)
+    def test_execution_times(testcase):
+        if float(testcase["time"]) > MAX_EXECUTION_TIME:
+            pytest.fail(
+                f"Gallery example {testcase['name']!r} from {testcase['file']!r}\n"
+                f"Took too long to run: Duration {testcase['time']}s > {MAX_EXECUTION_TIME}s",
+            )
```

Comment:
This includes a bunch of stuff that I would consider ancillary / unnecessary to answer the essential question, which I think is "how can I parse the XML to get something meaningful"? So if we're going to include a code block I think it should just answer that question, maybe in 3-4 lines. Maybe just show the maximum execution time for example?

Anyone doing `dev` work (which the SG consumers are doing) will be able to work from that hint to easily extend to their own use case that involves `Path`, `pytest`, `pytest.parametrize`, `pytest.fail` etc. which are really a bit too specific here.

If you want to have the 3-4 liner plus a permalink to the PyVista use case that's fine though